pkg/auth/acs_claims.go

```
@@ -79,7 +84,29 @@ func (c *ACSClaims) GetSubject() (string, error) {
 	return "", fmt.Errorf("can't find %q attribute in claims", tenantSubClaim)
 }
 
-// IsOrgAdmin ...
+// GetAudience returns the audience claim of the token. It identifies the token consumer.
+func (c *ACSClaims) GetAudience() ([]string, error) {
+	aud := make([]string, 0)
+	switch v := (*c)[audienceClaim].(type) {
+	case string:
+		aud = append(aud, v)
+	case []string:
+		aud = v
+	case []interface{}:
+		for _, a := range v {
+			vs, ok := a.(string)
+			if !ok {
+				return nil, fmt.Errorf("can't parse part of the audience claim: %q", a)
```

I'd rather fail open and create a log message that parts of the audience couldn't be parsed but not fail closed like we do here.

---

Done.